rust/src/index/vector/ivf.rs

```diff
@@ -133,10 +137,54 @@ impl std::fmt::Debug for IVFIndex {
     }
 }
 
+#[derive(Serialize)]
+pub struct IvfIndexPartitionStats {
+    index: usize,
+    length: u32,
+    offset: usize,
+    centroids: String,
+}
+
+#[derive(Serialize)]
+pub struct IvfIndexStats {
+    index_type: String,
+    uuid: String,
+    uri: String,
+    metric_type: String,
+    num_partitions: usize,
+    sub_index: String,
+    partitions: Vec<IvfIndexPartitionStats>,
+}
+
 impl Index for IVFIndex {
     fn as_any(&self) -> &dyn Any {
         self
     }
+
+    fn statistics(&self) -> Result<String> {
+        let partitions_stats = self
+            .ivf
+            .lengths
+            .iter()
+            .enumerate()
+            .map(|(i, &len)| IvfIndexPartitionStats {
+                index: i,
+                length: len,
+                offset: self.ivf.offsets[i],
+                centroids: format!("{:?}", self.ivf.centroids.value(i)),
+            })
+            .collect::<Vec<_>>();
+
+        Ok(serde_json::to_string(&IvfIndexStats {
+            index_type: "Ivf".to_string(),
+            uuid: self.uuid.clone(),
+            uri: to_local_path(self.reader.path()),
+            metric_type: self.metric_type.to_string(),
+            num_partitions: self.ivf.num_partitions(),
+            sub_index: self.sub_index.statistics()?,
```

Comment:
This would make a JSON-string within a JSON string, which will make parsing difficult. ideally, we want a user to be able to run `json.loads()` on the output string and the entire thing should be parsed.

Reply:
I think we can fix this by making `statistics()` return `serde_json::Value` instead. See my other comment.

Reply:
Done.